logic/src/commonTest/kotlin/com/wire/kalium/logic/data/call/CallRepositoryTest.kt

```diff
@@ -66,63 +93,100 @@ class CallRepositoryTest {
     }
 
     @Test
-    fun givenEmptyListOfCalls_whenGetAllCallsIsCalled_thenReturnAnEmptyListOfCalls() {
+    fun givenEmptyListOfCalls_whenGetAllCallsIsCalled_thenReturnAnEmptyListOfCalls() = runTest {
         val calls = callRepository.callsFlow()
 
-        assertEquals(0, calls.value.size)
+        calls.test {
+            assertEquals(0, awaitItem().size)
+        }
     }
 
     @Test
-    fun givenAListOfCallProfiles_whenGetAllCallsIsCalled_thenReturnAListOfCalls() {
+    fun givenAListOfCallProfiles_whenGetAllCallsIsCalled_thenReturnAListOfCalls() = runTest {
         callRepository.updateCallProfileFlow(CallProfile(mapOfCallProfiles))
 
         val calls = callRepository.callsFlow()
 
-        assertEquals(mapOfCallProfiles.size, calls.value.size)
-        assertTrue(calls.value[0].instanceOf(Call::class))
+        calls.test {
+            val list = awaitItem()
+            assertEquals(mapOfCallProfiles.size, list.size)
+            assertTrue(list[0].instanceOf(Call::class))
+        }
     }
 
     @Test
-    fun givenACallObject_whenCreateCallCalled_thenAddThatCallToTheFlow() {
+    fun givenACallObject_whenCreateCallCalled_thenAddThatCallToTheFlow() = runTest {
+        given(conversationRepository).suspendFunction(conversationRepository::getConversationDetailsById)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(ConversationDetails.Group(TestConversation.ONE_ON_ONE, LegalHoldStatus.ENABLED)))
+        given(userRepository).suspendFunction(userRepository::getKnownUser)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(TestUser.OTHER))
+        given(teamRepository).suspendFunction(teamRepository::getTeam)
+            .whenInvokedWith(any())
+            .thenReturn(flowOf(Team("team1", "team_1")))
         callRepository.updateCallProfileFlow(CallProfile(mapOf(startedCall.conversationId.toString() to startedCall)))
 
-        callRepository.createCall(answeredCall)
+        callRepository.createCall(conversationIdAnsweredCall, CallStatus.ANSWERED, "caller_id")
 
         val calls = callRepository.callsFlow()
-        assertEquals(2, calls.value.size)
-        assertEquals(calls.value[0], startedCall)
-        assertEquals(calls.value[1], answeredCall)
+
+        calls.test {
+            val list = awaitItem()
+            assertEquals(2, list.size)
+            assertEquals(list[0], startedCall)
+            assertEquals(list[1], answeredCall)
+        }
```

Comment:
we should verify that `getConversationDetailsById`, `getKnownUser` and `getTeam` are being invoked once